ext/webidl/00_webidl.js

```diff
@@ -835,7 +835,11 @@
 
   function createPromiseConverter(converter) {
     return (V, opts) =>
-      PromisePrototypeThen(PromiseResolve(V), (V) => converter(V, opts));
+      // should be able to handle thenables
+      // see: https://github.com/web-platform-tests/wpt/blob/a31d3ba53a79412793642366f3816c9a63f0cf57/streams/writable-streams/close.any.js#L207
+      typeof V?.then === "function"
+        ? PromisePrototypeThen(PromiseResolve(V), (V) => converter(V, opts))
+        : PromiseResolve(converter(V, opts));
```

Comment:
This change fixes the tests because the current logic (main) always adds an extra `Promise` by calling `.then`, see the following snippet

----

**This patch:**
```js
function promiseConverter(p) {
  return Promise.resolve(p);
}

const p1 = new Promise(resolve => resolve()).then(() => console.log('1'));
const p2 = promiseConverter(undefined).then(() => console.log('2'));
const p3 = new Promise(resolve => resolve()).then(() => console.log('3'));
// 1, 2, 3
```

**main**

```js
function promiseConverter(p) {
  return Promise.resolve(p).then(() => undefined);
}

const p1 = new Promise(resolve => resolve()).then(() => console.log('1'));
const p2 = promiseConverter(undefined).then(() => console.log('2'));
const p3 = new Promise(resolve => resolve()).then(() => console.log('3'));

// 1, 3, 2
```